### ISSUE TITLE
Credo.Check.Warning.WrongTestFileExtension displayed in non-test module in v1.7.4

### ISSUE THREAD

**Issue by jderr-mx** [rrrene/credo#1115]
### Environment

* Credo version (`mix credo -v`): 1.7.4
* Erlang/Elixir version (`elixir -v`): 1.16.0
* Operating system: MacOS Sonoma 14.2.1

### What were you trying to do?
I am using Neo Vim with coc.nvim and its coc-elixir plugin along with the ElixirLS language server
Starting with Credo 1.7.4 I see a warning next to the module name `Test files should end with .exs`
This warning is also generated when using the `--read-from-stdin` option on the cli
This warning disappears when downgrading to Credo 1.7.3

I was able to reproduce it by creating a new mix project named `a_project`
And then opening `lib/a_project.ex` in my IDE
Or running `mix credo --read-from-stdin < lib/a_project.ex`

### Actual outcome

<img width="772" alt="Screenshot 2024-02-16 at 9 36 46 AM" src="https://github.com/rrrene/credo/assets/16136290/7bf64ea1-ed08-487e-a1ae-98cdabcf8fd3">

<img width="1131" alt="Screenshot 2024-02-16 at 10 14 05 AM" src="https://github.com/rrrene/credo/assets/16136290/bdc4ba2c-2c22-4331-9a5b-353efaf61a99">

**Comment by rrrene** [comment]
I can reproduce this. Thx for reporting! :+1:

**Comment by rrrene** [comment]
@jderr-mx 
Again, thanks for reporting this. What a stupid regression 😞 It should now be fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by jderr-mx** [comment]
Hi @rrrene, I verified this fix is working in `master` 

Thank you 🎉

**Comment by scohen** [comment]
@rrrene this does seem to fix the issue, but the fix seems to have drastically slowed down credo's checks. What used to take 6 or  7 milliseconds now takes more than 50 (at which point lexical stops the plugin from running).

Can I take a stab at improving the performance before you make a release? Releasing main right now will break lexical's credo plugin for whomever uses it (and people won't have credo warnings in their editors any more).

**Comment by scohen** [comment]
@rrrene you can ignore my comment, the slowdown was only for the first run (likely due to the `Regex.compile`), subsequent runs are back to the single digit milliseconds, which is plenty fast for lexical. 

Can we get a release soon?

**Comment by RobertoSchneiders** [comment]
Thanks for fixing this, I'm waiting for the release as well.

**Comment by rrrene** [comment]
@jderr-mx @scohen @RobertoSchneiders This is live in `v1.7.5`.

**Comment by scohen** [comment]
Thanks, @rrrene! We love credo.
